app/services/preserved_object_handler.rb

```diff
@@ -56,115 +28,110 @@ def initialize(druid, incoming_version, incoming_size, endpoint)
     @incoming_version = version_string_to_int(incoming_version)
     @incoming_size = string_to_int(incoming_size)
     @endpoint = endpoint
+    @handler_results = PreservedObjectHandlerResults.new(druid, incoming_version, incoming_size, endpoint)
   end
 
   def create_after_validation
-    results = []
     if invalid?
-      results << result_hash(INVALID_ARGUMENTS, errors.full_messages)
+      handler_results.add_result(PreservedObjectHandlerResults::INVALID_ARGUMENTS, errors.full_messages)
     elsif PreservedObject.exists?(druid: druid)
-      results << result_hash(OBJECT_ALREADY_EXISTS, 'PreservedObject')
+      handler_results.add_result(PreservedObjectHandlerResults::OBJECT_ALREADY_EXISTS, 'PreservedObject')
     elsif moab_validation_errors.empty?
-      results.concat(create_db_objects(Status.default_status, true))
+      create_db_objects(Status.default_status, true)
     else
-      results.concat(create_db_objects(Status.invalid_moab, true))
+      create_db_objects(Status.invalid_moab, true)
     end
 
-    log_results(results)
-    results
+    handler_results.log_my_results
```

Comment:
`log_my_results` is a hokey name;  i wanted to distinguish it from the class `log_results` method ... suggestions welcome.

Reply:
i think it'd be reasonable to collapse the two into an instance method called `log_results`, something like:
```ruby
def log_results
  result_array.each do |r|
    severity = logger_severity_level(r.keys.first)
    msg = r.values.first
    Rails.logger.log(severity, msg)
  end
end
```